projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	url        = "http://localhost:8080"
+	maxRetries = 3
+)
+
+func main() {
+
+	// Create an HTTP client
+	client := &http.Client{}
+	// This loop will run only if the number of reties doesn't exceed 3
+	for retries := 0; retries < maxRetries; retries++ {
```

Comment:
What will happen if the server _always_ succeeds (i.e. every response it gives is a 200)?

Reply:
Since I didn't implement `os.Exit(0)` yet , the loop doesn't stop on success. I'm trying to workout how to implement it since I should only call it from `main` :)